pkg/collector/container.go

```diff
@@ -33,6 +35,22 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 		image = cfg.CollectorImage()
 	}
 
+	// build port map
+	ports := getConfigContainerPorts(logger, otelcol.Spec.Config)
+	for _, p := range otelcol.Spec.ContainerPorts {
+		if len(p.Name) > 15 {
+			p.Name = p.Name[:15]
```

Comment:
Maybe we should log something here for the operator to make clear what is being truncated